includes.php

```diff
@@ -250,38 +264,45 @@ function hookpress_register_hooks() {
 
 function hookpress_generic_action($id,$args) {
 	global $hookpress_version, $wpdb, $hookpress_actions, $hookpress_filters, $wp_version;
-	
+
 	$webhooks = hookpress_get_hooks( );
 	$desc = $webhooks[$id];
 
 	do_action( 'hookpress_hook_fired', $desc );
 
 	$obj = array();
-	
+
 	// generate the expected argument names
 	if (isset($desc['type']) && $desc['type'] == 'filter')
-		$arg_names = $hookpress_filters[$desc['hook']];	
-	else
-		$arg_names = $hookpress_actions[$desc['hook']];
-	
+		$arg_names = $hookpress_filters[$desc['hook']];
+	else {
+		$hooks = apply_filters( 'hookpress_actions', $hookpress_actions );
+		$arg_names = $hooks[$desc['hook']];
+	}
+
 	foreach($args as $i => $arg) {
-		$newobj = array();
+		$newobj = apply_filters('hookpress_get_data', $obj, $arg_names[$i], $arg);
+
 		switch($arg_names[$i]) {
 			case 'POST':
 			case 'ATTACHMENT':
 				$newobj = get_post($arg,ARRAY_A);
 
-				if ($arg_names[$i] == 'POST')
+				if ($arg_names[$i] == 'POST') {
 					$newobj["post_url"] = get_permalink($newobj["ID"]);
-					
+
+					$meta = get_post_meta($newobj["ID"], '', true);
+					$newobj = array_merge($meta, $newobj);
```

Comment:
This too goes with the above comment -- adding extra columns should be a separate patch.